src/edge.hal-interface/src/hal_interface.py

```diff
@@ -492,26 +492,28 @@ async def start(self):
             # Make the application run forever
             await asyncio.sleep(1000)
 
-    async def init_message_broker(self):
-        self.broker = NamedMqttBroker('HalInterface-{}'.format(id(self)), self.config['mqtt']['connectionString'], self.config['mqtt']['ns'] or 'hal/')
+    async def init_protocol_gateway(self):
+        self.protocol_gateway = ProtocolGateway(self.config['protocolGateway'], 'HalInterface-{}'.format(id(self)), True)
+        if len(self.config['protocolGateway']['adapters']) is not 1:
+            raise Exception('Invalid ProtocolGateway Configuration: only one adapter is allowed!')
 
         # pylint: disable=anomalous-backslash-in-string
-        self.topic_regex = re.compile('^{}([^\.]+)\.(.+)\/(?:enable|disable)$'.format((self.broker.topic_ns or '').replace('\\/', '\\\\/')))
-        self.logger.info("mqtt broker created")
+        self.topic_regex = re.compile('([^\.]+)\.(.+)\/(?:enable|disable)$')
```

Comment:
Should the topic namespace not be considered anymore?

Reply:
The callbacks are called with the namespace stripped off. It is not passed into subscribe and publish methods, so not got in the callbacks.